core/src/main/java/io/confluent/rest/Application.java

```diff
@@ -197,20 +200,18 @@ public Server createServer() throws ServletException {
     // The configuration for the JAX-RS REST service
     ResourceConfig resourceConfig = new ResourceConfig();
 
+    Map<String, String> combinedMetricsTags = new HashMap<>();
+    for (ApplicationContext ctx : appCtx) {
+      combinedMetricsTags.putAll(ctx.getMetricsTags());
```

Comment:
Is it possible two different context may try to add metrics with the same key? If so, do we want to error out or at least log some warning?

Reply:
It will actually fail if you have duplicate tags on registration.

Reply:
Ill add a test though good thinking

Reply:
Actually you are right since I'm adding these tags before registration. Good catch fixed